cmake/snmalloc.cmake

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+if(USE_SNMALLOC)
+  set(SNMALLOC_BUILD_TESTING OFF)
+  set(SNMALLOC_STATIC_LIBRARY_PREFIX "")
+  add_subdirectory(3rdparty/exported/snmalloc EXCLUDE_FROM_ALL)
+
+  if(PACKAGING)
+    # We move snmalloc includes under 3rdparty/ when installing ccf, so
+    # overwrite the existing value with what we want.
+    set_target_properties(snmalloc PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "")
```

Comment:
If we move snmalloc to 3rdparty/internal/, then we stop exporting the headers, which is I think what we want?